gfw_pixetl/pixetl_prep.py

```diff
@@ -3,35 +3,66 @@
 
 import click
 
+from gfw_pixetl import get_module_logger
 from gfw_pixetl.sources import RasterSource
 from gfw_pixetl.utils import get_bucket, upload_geometries
 from gfw_pixetl.utils.aws import get_aws_files
 from gfw_pixetl.utils.google import get_gs_files
 from gfw_pixetl.utils.utils import DummyTile
+from concurrent.futures import ProcessPoolExecutor, as_completed
+
+LOGGER = get_module_logger(__name__)
 
 
 def get_key_from_vsi(vsi_path: str) -> str:
     key = vsi_path.split("/")[3:]
     return "/".join(key)
 
 
+def parallel_raster_source(uri) -> RasterSource:
+    return RasterSource(uri)
+
+
+def parallel_get_tiles(files) -> List[DummyTile]:
+    '''Given a potentially large list of tiles files, return the list of
+    DummyTile/RasterSource objects associated with each tile. Creating the
+    RasterSource objects can be slow, since it requires fetching meta-data for
+    each tile file (on S3).
+    '''
+
+    future_tiles = {}
+    tiles: List[DummyTile] = list()
+
+    with ProcessPoolExecutor(max_workers=16) as executor:
```

Comment:
Rather than hard-code to 16, change this to use GLOBALS.num_processes as here:
https://github.com/wri/gfw_pixetl/blob/master/gfw_pixetl/pipes/raster_pipe.py#L60
This is meant to scale with the instance size.

Reply:
👍